src/social-network-provider/twitter.com/ui/fetch.ts

```diff
@@ -0,0 +1,155 @@
+import Services from '../../../extension/service'
+import { PersonIdentifier } from '../../../database/type'
+import { host } from '../index'
+import { getEmptyPostInfo, SocialNetworkUI } from '../../../social-network/ui'
+import { deconstructPayload } from '../../../utils/type-transform/Payload'
+import { regexMatch, timeout } from '../../../utils/utils'
+import { hasPostPopup } from '../utils/status'
+import { equal, notEmpty, notNullable } from '../../../utils/assert'
+
+export const resolveLastRecognizedIdentity = (self: SocialNetworkUI) => {
+    const selfSelector = selfInfoSelectors().screenName
+    const assign = () => {
+        try {
+            const ref = self.lastRecognizedIdentity
+            const info = selfInfoSelectors().screenName.evaluate()
+            const id = new PersonIdentifier(host, notEmpty(info, 'user id not found')[0])
+            equal(id.isUnknown, false, 'user id not recognized')
+            ref.value = { identifier: id }
+        } catch (e) {
+            /**
+             * This function expects to fail to collect info,
+             * maybe don't throw is a good idea
+             *                                  -- Jack-Works
+             */
+            console.log(e)
+        }
+    }
+    new MutationObserverWatcher(selfSelector)
+        .enableSingleMode()
+        .addListener('onAdd', () => assign())
+        .addListener('onChange', () => assign())
+        .startWatch()
+        .then()
+}
+
+export const resolveInfoFromBioCard = () => {
+    const userAvatarUrl = notNullable(
+        bioCard()
+            .nth(0)
+            .querySelector<HTMLImageElement>('img')
+            .evaluate(),
+    ).src
+    const userNames = notNullable(
+        bioCard()
+            .nth(1)
+            .evaluate(),
+    ).innerText.split('\n')
+    const userBio = notNullable(
+        bioCard()
+            .nth(2)
+            .evaluate(),
+    ).innerText
+    return {
+        userAvatarUrl,
+        userName: userNames[0],
+        userScreenName: userNames[1],
+        userBio,
+    }
+}
+
+const registerBioCollector = () => {
+    // This object will not be garbage collected
+    new MutationObserverWatcher(bioCard())
+        .enableSingleMode()
+        .useForeach(node => {
+            const refreshUserInfo = () => {
+                const r = resolveInfoFromBioCard()
+                const text = node.innerText
+                const p = new PersonIdentifier(host, r.userScreenName)
+                Services.Crypto.verifyOthersProve(text, p).then()
+                Services.People.updatePersonInfo(p, {
+                    nickname: r.userName,
+                    avatarURL: r.userAvatarUrl,
+                }).then()
+            }
+            refreshUserInfo()
+            return {
+                onNodeMutation: refreshUserInfo,
+                onTargetChanged: refreshUserInfo,
+            }
+        })
+        .startWatch()
+        .then()
```

Comment:
`then()` on the Watcher have side effects. It implments `PromiseLike` but it is not a Promise.